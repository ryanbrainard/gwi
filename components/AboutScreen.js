import React from "react";
import {
  Linking,
  Image,
  StyleSheet,
  Text,
  TouchableOpacity,
  View
} from "react-native";
import { Segment } from "expo";
import config from "../config";
import { DimensionsConsumer } from "./DimensionsContext";
import appJson from "../app.json";

export default class AboutScreen extends React.Component {
  static navigationOptions = {
    title: "About"
  };

  render() {
    return (
      <DimensionsConsumer>
        {({ window: { width, height }, isPortrait }) =>
          this.renderInternal(Math.min(width, height), isPortrait)
        }
      </DimensionsConsumer>
    );
  }

  renderInternal(size, isPortrait) {
    const logoSize = size / 2.5;

    const LinkButton = ({ text, href }) => (
      <TouchableOpacity
        style={styles.button}
        onPress={() => {
          Linking.openURL(href);
          Segment.trackWithProperties("about-link-button-press", {
            text,
            href
          });
        }}
      >
        <Text style={styles.buttonText}>{text}</Text>
      </TouchableOpacity>
    );

    return (
      <View style={styles.container}>
        <Image
          source={require("../assets/images/gwi-logo-green-trans-1200.png")}
          style={{ width: logoSize, height: logoSize }}
        />
        <Text style={styles.titleText}>{appJson.expo.name}</Text>
        <Text>Made in cooperation with Motivate Korean</Text>
        <Text>Apache License 2.0 Open Source Licensed</Text>
        <Text>Â© 2018 Ryan Brainard</Text>
        <Text>v{appJson.expo.version}</Text>
        <Text />
        <View
          style={[
            styles.bodyContainer,
            { flexDirection: isPortrait ? "column" : "row" }
          ]}
        >
          <LinkButton
            text="Korean Coaching"
            href="http://www.motivatekorean.com/hire-us"
          />
          <LinkButton
            text="Report an Issue"
            href={`${appJson.expo.githubUrl}/issues`}
          />
          <LinkButton text="Source Code" href={appJson.expo.githubUrl} />
        </View>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: "center",
    alignItems: "center"
  },
  bodyContainer: {
    flex: 1,
    alignItems: "center"
  },
  titleText: {
    fontSize: 24,
    fontWeight: "bold"
  },
  button: {
    backgroundColor: config.colors.about.primary,
    borderRadius: 75,
    padding: 12,
    justifyContent: "center",
    margin: 10
  },
  buttonText: {
    fontWeight: "bold",
    fontSize: 14,
    color: config.colors.text
  }
});
