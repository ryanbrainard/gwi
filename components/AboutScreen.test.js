import React from "react";
import AboutScreen from "./AboutScreen";
import renderer from "react-test-renderer";

test("renders correctly", () => {
  const tree = renderer.create(<AboutScreen />).toJSON();
  expect(tree).toMatchSnapshot();
});
