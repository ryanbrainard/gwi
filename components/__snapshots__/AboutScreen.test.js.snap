// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Image
    source={1}
    style={
      Object {
        "height": 300,
        "width": 300,
      }
    }
  />
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "fontSize": 24,
        "fontWeight": "bold",
      }
    }
  >
    Korean Ear
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    Made in cooperation with Motivate Korean
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    Open Source Apache License 2.0
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    Â© 2018 Ryan Brainard
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    v
    0.2.7
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  />
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flex": 1,
        },
        Object {
          "flexDirection": "column",
        },
      ]
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#3CB371",
          "borderRadius": 75,
          "justifyContent": "center",
          "margin": 10,
          "opacity": 1,
          "padding": 12,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#000000",
            "fontSize": 14,
            "fontWeight": "bold",
          }
        }
      >
        Korean Coaching
      </Text>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#3CB371",
          "borderRadius": 75,
          "justifyContent": "center",
          "margin": 10,
          "opacity": 1,
          "padding": 12,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#000000",
            "fontSize": 14,
            "fontWeight": "bold",
          }
        }
      >
        Report an Issue
      </Text>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#3CB371",
          "borderRadius": 75,
          "justifyContent": "center",
          "margin": 10,
          "opacity": 1,
          "padding": 12,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#000000",
            "fontSize": 14,
            "fontWeight": "bold",
          }
        }
      >
        Source Code
      </Text>
    </View>
  </View>
</View>
`;
